src/server/utils/zitadel.ts

```diff
@@ -0,0 +1,173 @@
+import {Request} from '@gravity-ui/expresskit';
+import {AppContext} from '@gravity-ui/nodekit';
+import axios from 'axios';
+import axiosRetry from 'axios-retry';
+import NodeCache from 'node-cache';
+import {Issuer, TokenSet} from 'openid-client';
+
+import {
+    clientId,
+    clientSecret,
+    serviceClientId,
+    serviceClientSecret,
+    zitadelProjectId,
+    zitadelUri,
+} from '../app-env';
+import {getDuration} from '../components/charts-engine/components/utils';
+
+const cache = new NodeCache();
+
+axiosRetry(axios, {
```

Comment:
It's retries for global `axios` instance. Need to set retries only for these requests.

Reply:
discussed offline, fixed for introspection